homeassistant/components/mqtt/__init__.py

```diff
@@ -532,10 +533,21 @@ def async_subscribe(self, topic, qos):
         This method is a coroutine.
         """
         if not isinstance(topic, str):
-            raise HomeAssistantError("topic need to be a string!")
+            raise HomeAssistantError("topic needs to be a string!")
+
+        _LOGGER.debug("Subscribing to %s", topic)
 
         with (yield from self._paho_lock):
-            if topic in self.subscribed_topics:
+            # Don't re-subscribe to wanted subscription topics.
+            if topic in self.wanted_topics:
```

Comment:
I'm using `wanted_topics` here, because it wouldn't make any sense to call the `paho-mqtt` subscribe method as it will soon be called anyway.